corehq/apps/data_dictionary/views.py

```diff
@@ -47,45 +64,78 @@ def data_dictionary_json(request, domain, case_type_name=None):
 @toggles.DATA_DICTIONARY.required_decorator()
 def update_case_property(request, domain):
-    case_type = request.POST.get('caseType')
-    name = request.POST.get('name')
-    description = request.POST.get('description')
-    data_type = request.POST.get('data_type')
-    try:
-        prop = CaseProperty.objects.get(
-            name=name, case_type__name=case_type, case_type__domain=domain
+    property_list = json.loads(request.POST.get('properties'))
+    for property in property_list:
+        case_type = property.get('caseType')
+        name = property.get('name')
+        description = property.get('description')
+        data_type = property.get('data_type')
+        group = property.get('group')
+        prop = CaseProperty.get_or_create(
+            name=name, case_type=case_type, domain=domain
         )
-    except CaseProperty.DoesNotExist:
-        return JsonResponse({
-            'status': "failed",
-            "error": "property does not exist"
-        }, status=404)
-    if data_type:
-        prop.data_type = data_type
-    if description:
-        prop.description = description
-    try:
-        prop.full_clean()
-    except ValidationError as e:
-        return JsonResponse({"status": "failed", "error": unicode(e)}, status=400)
-    prop.save()
+        if data_type:
+            prop.data_type = data_type
+        if description:
+            prop.description = description
+        if group:
+            prop.group = group
+        try:
+            prop.full_clean()
+        except ValidationError as e:
+            return JsonResponse({"status": "failed", "error": unicode(e)}, status=400)
+        prop.save()
     return JsonResponse({"status": "success"})
 
 
-class DataDictionaryView(BaseDomainView):
+def export_data_dictionary(request, domain):
+    queryset = CaseType.objects.filter(domain=domain).prefetch_related(
+        Prefetch('properties', queryset=CaseProperty.objects.order_by('name'))
+    )
+    export_data = {}
+    for case_type in queryset:
+        export_data[case_type.name] = [{
+            'Case Property': prop.name,
+            'Group': prop.group,
+            'Data Type': prop.data_type,
+            'Description': prop.description,
+        } for prop in case_type.properties.all()]
+    headers = ('Case Property', 'Group', 'Data Type', 'Description')
+    outfile = StringIO()
+    writer = Excel2007ExportWriter()
+    header_table = [(tab_name, [headers]) for tab_name in export_data]
+    writer.open(header_table=header_table, file=outfile)
+    for tab_name, tab in export_data.items():
+        tab_rows = []
+        for row in tab:
+            tab_rows.append([row.get(header, '') for header in headers])
+        writer.write([(tab_name, tab_rows)])
+    writer.close()
+    response = HttpResponse(content_type=Format.from_format('xlsx').mimetype)
+    response['Content-Disposition'] = 'attachment; filename="data_dictionary.xlsx"'
+    response.write(outfile.getvalue())
```

Comment:
perhaps not that important in this case but could get nginx to handle the file transfer like this: https://github.com/dimagi/commcare-hq/blob/1036f963bd4ef6d47131010bd6f71024d6391bf1/corehq/util/zip_utils.py#L54-L54